opm/output/data/Aquifer.hpp

```diff
@@ -37,15 +37,40 @@ namespace Opm { namespace data {
 
     struct FetkovichData
     {
-        double initVolume;
-        double prodIndex;
-        double timeConstant;
+        double initVolume{};
+        double prodIndex{};
+        double timeConstant{};
+
+        bool operator==(const FetkovichData& other) const;
+
+        // MessageBufferType API should be similar to Dune::MessageBufferIF
+        template <class MessageBufferType>
+        void write(MessageBufferType& buffer) const;
+
+        // MessageBufferType API should be similar to Dune::MessageBufferIF
+        template <class MessageBufferType>
+        void read(MessageBufferType& buffer);
     };
 
     struct CarterTracyData
     {
-        double dimensionless_time;
-        double dimensionless_pressure;
+        double timeConstant{};
+        double influxConstant{};
+        double waterDensity{};
+        double waterViscosity{};
+
+        double dimensionless_time{};
+        double dimensionless_pressure{};
+
+        bool operator==(const CarterTracyData& other) const;
+
+        // MessageBufferType API should be similar to Dune::MessageBufferIF
+        template <class MessageBufferType>
+        void write(MessageBufferType& buffer) const;
+
+        // MessageBufferType API should be similar to Dune::MessageBufferIF
+        template <class MessageBufferType>
+        void read(MessageBufferType& buffer);
```

Comment:
We should create a mock `MessageBufferType` with the correct API so that these `read()` and `write()` methods can be tested.

Reply:
> We should create a mock `MessageBufferType` with the correct API

We have one in [`test_Summary`](https://github.com/OPM/opm-common/blob/dc207ae8c5e13877388ffdfe36ea6b1724a36266/tests/test_Summary.cpp#L1872).  I've pushed an update that adds such a unit test.